apps/backend/src/libs/modules/token/token.ts

```diff
@@ -0,0 +1,11 @@
+import { config } from "~/libs/modules/config/config.js";
+
+import { BaseToken } from "./base-token.module.js";
+
+const token = new BaseToken({
```

Comment:
```suggestion
const token = new BaseToken<TokenPayload>({
```
Create a new type

Reply:
should I export this TokenPayload type from token.ts?

Reply:
If I export it from token.ts, the code breaks

Reply:
You should create TokenPayload in shared folder and use it in Backend. Not sure about exporting from token.ts, if you don't use that export, then you shouldn't export from there

Reply:
I've encountered a problem.

When I move the type definition to shared package, i can't extend TokenPayload interface to Include JWTPayload, since "jose" package is not present in shared.
<img width="673" alt="Screenshot 2024-08-20 at 10 44 00" src="https://github.com/user-attachments/assets/1990d791-b141-4abc-9c88-357bf2c47f3f">

Because of that I cannot use TokenPayload in the generic slot for BaseToken constructor call:

<img width="743" alt="Screenshot 2024-08-20 at 10 45 24" src="https://github.com/user-attachments/assets/a697b7f5-b534-446e-a5d3-2afa9a75d74c">

This happens because generic in BaseToken definition extends the JWTPayload. I cannot delete this generic extension since jwtVerify in decode method of BaseToken class requires a JWTPayload as an argument:

<img width="792" alt="Screenshot 2024-08-20 at 10 48 41" src="https://github.com/user-attachments/assets/d8c60a55-cd5d-42c9-9908-c7e2832bdd62">

first solution that comes to mind is to install jose package in shared, but previously in previous comments it was stated that jose should be present in backend only. Are there any other common approaches for situations like this one?

Reply:
Try this:
```
class BaseToken<T extends JWTPayload>
```

Reply:
I have this in the source code, but removed it from the screenshot so that the type conflict is highlited by IDE